File: day-1/dart/calories.dart
import 'dart:async';
import 'dart:io';
import 'dart:convert';

main() async {
  var lines = await File("inputs.txt")
      .openRead()
      .transform(utf8.decoder)
      .transform(LineSplitter());
  var highest1 = 0;
  var highest2 = 0;
  var highest3 = 0;
  var sum = 0;
  await lines.forEach((line) {
    if (line == "") {
      if (sum > highest1) {
        highest3 = highest2;
        highest2 = highest1;
        highest1 = sum;
      } else if (sum > highest2) {
        highest3 = highest2;
        highest2 = sum;
      } else if (sum > highest3) {
        highest3 = sum;
      }
      sum = 0;
    } else {
      var num = int.parse(line);
      sum += num;
    }
  });

  if (sum > highest1) {
    highest3 = highest2;
    highest2 = highest1;
    highest1 = sum;
  } else if (sum > highest2) {
    highest3 = highest2;
    highest2 = sum;
  } else if (sum > highest3) {
    highest3 = sum;
  }

  print("Highest is: $highest1");
  print("Total of highest 3: ${highest1 + highest2 + highest3}");
}

File: day-1/lua/calories.lua


local highest1 = 0
local highest2 = 0
local highest3 = 0
local sum = 0



for line in io.lines("inputs.txt") do 
  if line == "" then
    if sum > highest1 then
      highest3 = highest2
      highest2 = highest1
      highest1 = sum
    elseif sum > highest2 then
      highest3 = highest2
      highest2 = sum
    elseif sum > highest3 then
      highest3 = sum
    end
    sum = 0
  else
    local num = tonumber(line)
    sum = sum + num
  end
end

if sum > highest1 then
  highest3 = highest2
  highest2 = highest1
  highest1 = sum
elseif sum > highest2 then
  highest3 = highest2
  highest2 = sum
elseif sum > highest3 then
  highest3 = sum
end

print("Highest is:", highest1)
print("Total of highest 3:", highest1 + highest2 + highest3)